apps/token-manager/app/src/components/CustomLabelIdentityBadge/CustomLabelIdentityBadge.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { Badge, IdentityBadge, font } from '@aragon/ui'
+import { CustomLabelModalContext } from '../CustomLabelModal/CustomLabelModalManager'
+import { IdentityContext } from '../IdentityManager/IdentityManager'
+
+const CustomLabelIdentityBadge = ({ address, ...props }) => {
+  const { resolve } = React.useContext(IdentityContext)
+  const { showCustomLabelModal, updates$ } = React.useContext(
+    CustomLabelModalContext
+  )
+  const [label, setLabel] = React.useState()
+  const handleResolve = async () => {
+    try {
+      const { name = null } = await resolve(address)
+      setLabel(name)
+    } catch (e) {
+      // address does not ressolve to identity
+    }
+  }
+  const handleClick = () => {
+    showCustomLabelModal(address)
+      .then(handleResolve)
+      .catch(e => {
+        /* user cancelled modify intent */
+      })
+  }
+  React.useEffect(() => {
+    handleResolve(address)
+    const subscription = updates$.subscribe(addr => {
+      if (addr === address) {
+        handleResolve(address)
+      }
+    })
+    return () => subscription.unsubscribe()
+  }, [])
```

Comment:
Should we add `address` to the array passed in the second argument? This could prevent redundant resolutions.

Reply:
This effect only happens once:

> If you want to run an effect and clean it up only once (on mount and unmount), you can pass an empty array ([]) as a second argument.

[Hooks Effect](https://reactjs.org/docs/hooks-effect.html)

Reply:
Thanks that clarifies it